server/src/test/java/org/apache/druid/server/coordinator/rules/BroadcastDistributionRuleTest.java

```diff
@@ -253,6 +313,46 @@ public void testBroadcastToSingleDataSource()
     assertFalse(holderOfSmallSegment.getPeon().getSegmentsToLoad().contains(smallSegment));
   }
 
+  /**
+   * Servers:
+   * name         | segments
+   * -------------+--------------
+   * general      | large segment
+   * maintenance1 | small segment
+   * maintenance2 | large segment
+   *
+   * After running the rule for the small segment:
+   * general      | large & small segments
+   * maintenance1 |
+   * maintenance2 | large segment
+   */
+  @Test
+  public void testBroadcastWithMaintenance()
+  {
+    final ForeverBroadcastDistributionRule rule = new ForeverBroadcastDistributionRule(ImmutableList.of("large_source"));
+
+    CoordinatorStats stats = rule.run(
+        null,
+        DruidCoordinatorRuntimeParams.newBuilder()
+                                     .withDruidCluster(secondCluster)
+                                     .withSegmentReplicantLookup(SegmentReplicantLookup.make(secondCluster))
+                                     .withBalancerReferenceTimestamp(DateTimes.of("2013-01-01"))
+                                     .withAvailableSegments(Lists.newArrayList(
```

Comment:
Could add `withAvailableSegments(DataSegment... segments)` method for convenience.

Reply:
I don't think we should extend a production class method list for test purposes only.

Reply:
This is what should be done. Test writing convenience and succinctness is important. There are a lot of methods marked as `@VisibleForTesting` in production classes.

Reply:
Sure, but imagine, for example, given a production class with a single method, I add 3 methods which are used only in tests, the test will look more succinct, but the class won't. The coder who needs to understand the logic of the whole component needs to check usages of all 4 methods and scan through them to confirm that those 3 used only in tests and are not called in production and shouldn't be followed to understand how the component operates, waste of time and concentration, I think.

Reply:
:white_check_mark: